common/models.py

```diff
@@ -226,13 +227,19 @@ class AssignedTask(models.Model):
     clazz = models.ForeignKey(Class, on_delete=models.CASCADE)
     assigned = models.DateTimeField()
     deadline = models.DateTimeField(null=True, blank=True)
-    hard_deadline = models.BooleanField(null=True, blank=True)
+    hard_deadline = models.BooleanField(blank=True, null=False, default=False)
```

Comment:
I think that we can omit the `blank` attribute, it should only affect the admin (because we don't use Django forms), and since it's a boolean, it cannot be "not filled". `null=False` is also the default, so `default=False` should be enough.